### DISCUSSION TITLE
Support Velocity Proxy, Waterfall will be replaced by Velocity by the paper team

### DISCUSSION THREAD

**Discussion by TomLewis** [Zrips/CMI#6382] [Enhancement]
I cant find the original post, but the velocity proxy is now under the paper organisation.
https://github.com/PaperMC/Velocity/commit/2586210ca67f2510eb4f91bf7567643f8a26ee7b

This is because paper are phasing out waterfall and replacing it with velocity. can you please make sure you support velocity asap for the proxy integration (and maybe have a spigot page for it too like you do with the libs now) instead of just a link to your wordpress download!

**Comment by Baktus79** [comment]
I think Velocity is a proxy that will be more important than before in the future, and there are several servers that are considering switching to this. The one thing that makes my choice difficult to switch to Velocity is precisely CMIB. This is a discussion that has been going on here for a while. How about making CMIB open so that others can take care of the job of porting it to Velocity?

**Comment by radiant-ai** [comment]
Velocity offers support for legacy plugin messaging and it should be easy to port CMIB to Velocity. The proxy is superior in performance and features so everyone considers switching to it. Please let us port CMIB ourselves or do it yourself.

**Comment by hamcoremusic** [comment]
Please include Velocity support! Would be really cool to have and will make the switchover really easy.

**Comment by Besoj** [comment]
Please include Velocity support! Would be very beneficial to the community

**Comment by 1ntent1on** [comment]
Agree, it would be really great to have that option.

**Comment by xiaoxing365** [comment]
Some developers have made CMI compatible with Velocity
GITHUB: https://github.com/pintux98/CMIV
SPIGOTMC: https://www.spigotmc.org/resources/cmi-velocity.111751/

**Comment by mrfloris** [comment]
https://www.spigotmc.org/resources/cmi-298-commands-insane-kits-portals-essentials-economy-mysql-sqlite-much-more.3742/update?update=521168

There is a mention in the changelog for velocity support in today's CMI release.
You can get the free cmiv jar file here: https://www.zrips.net/cmiv/

**Comment by TomLewis** [comment]
Oh hell yeah!
